sklearn/tree/export.py

```diff
@@ -287,107 +303,422 @@ def node_to_str(tree, node_id, criterion):
         node_string += value_text + characters[4]
 
         # Write node majority class
-        if (class_names is not None and
+        if (self.class_names is not None and
                 tree.n_classes[0] != 1 and
                 tree.n_outputs == 1):
             # Only done for single-output classification trees
             if labels:
                 node_string += 'class = '
-            if class_names is not True:
-                class_name = class_names[np.argmax(value)]
+            if self.class_names is not True:
+                class_name = self.class_names[np.argmax(value)]
             else:
                 class_name = "y%s%s%s" % (characters[1],
                                           np.argmax(value),
                                           characters[2])
             node_string += class_name
 
         # Clean up any trailing newlines
-        if node_string[-2:] == '\\n':
-            node_string = node_string[:-2]
-        if node_string[-5:] == '<br/>':
-            node_string = node_string[:-5]
+        if node_string.endswith(characters[4]):
+            node_string = node_string[:-len(characters[4])]
 
         return node_string + characters[5]
 
-    def recurse(tree, node_id, criterion, parent=None, depth=0):
+
+class _DOTTreeExporter(_BaseTreeExporter):
+    def __init__(self, out_file=SENTINEL, max_depth=None,
+                 feature_names=None, class_names=None, label='all',
+                 filled=False, leaves_parallel=False, impurity=True,
+                 node_ids=False, proportion=False, rotate=False, rounded=False,
+                 special_characters=False, precision=3):
+        self.out_file = out_file
```

Comment:
I'd appreciate making use of a super `__init__` so that this is a refactored as possible. Given that we don't extensively test the MPL exporter, I feel this is necessary to minimize the amount of MPL export code that we manually review for bugs (on the assumption that it is factored into both implementations). Ideally don't have any default values for the super init params, so that there's no risk of not passing something.

Reply:
will do. I'll try to prioritize the 0.19.1 bugfixes but it looks like you got a bunch of them reviewed already :)